PORTING.md

```diff
@@ -0,0 +1,354 @@
+
+### Adapt the startup- and linkerscripts
+
+The [Cortex-M startup script](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/driver/core/cortex/startup.c.in?ts=4) is pretty universal. These are the steps it does:
+
+1. Generate the exception table and places it at the beginning of Flash.
+2. Fill the stack with magic values to catch unintentional stack execution.
+3. Call platform specific code (enabling internal memories, **needs porting**).
+4. Copy and zero internal memories.
+5. Initialize the Cortex-M CPU.
+6. Call `xpcc_gpio_enable()` to enable clock to GPIO (**needs porting**).
+7. Call external hardware initialize hook.
+8. Copy and zero external memories.
+9. Initialize heap on all memories.
+10. Call libc init incl. execution of C++ static constructors.
+11. Call `main()`.
+
+We'll write the `xpcc_gpio_enable()` function when we port the GPIO driver.
+
+To provide code for step 3, copy and adapt [the macro file for the STM32 devices](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/driver/core/cortex/stm32/stm32.macros?ts=4).
+You can leave the `defines()` and `appendInterrupt()` macros as they are, but you need to implement the `startupCode()` macro with whatever platform specific hardware needs to be initialized. Don't do complicated stuff here, the system is not fully booted yet!
+
+The [simplest linkerscript available is `stm32/stm32_ram.ld.in`](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/driver/core/cortex/stm32/stm32_ram.ld.in?ts=4), which fits everything into one SRAM region.
+The linkerscipt diagram in the comment gives you an overview of what goes where, check that against the devices memory map. Note that while the diagram shows specific addresses, the section addresses and sizes are taken from the device file!
+
+You need to tell the [core driver which linkerscript to use for you device](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/driver/core/cortex/driver.xml?ts=4#L29-L40):
+```xml
+<template device-platform="sam" out="linkerscript.ld">stm32/stm32_ram.ld.in</template>
+```
+
+### Compile a minimal example
+
+At this point you should test your changes to the build system and the port of the core driver.
+Create an empty example in `xpcc/examples/sam4/minimal`, containing:
+
+- the default `SConstruct` file:
+```python
+# path to the xpcc root directory
+xpccpath = '../../..'
+# execute the common SConstruct file
+execfile(xpccpath + '/scons/SConstruct')
+```
+
+- an empty `main.cpp` file:
+```cpp
+// your copyright with bsd license
```

Comment:
why not just include the short BSD license header that we use, including the Copyright template with `{{ your_name }}`?

I always find it tedious to get hold of this.